### PR TITLE
refactor #1501: migrated invoice usecase to kotlin

### LINKED ISSUE #1501: refactor: migrate invoice usecase tot kotlin

### PR DESCRIPTION
**Pull request by PratyushSingh07**
## Issue Fix
Fixes #1501 

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/invoice/FetchInvoices.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ */
+class FetchInvoices @Inject constructor(private val mFineractRepository: FineractRepository) :
+    UseCase<FetchInvoices.RequestValues?, FetchInvoices.ResponseValue?>() {
+
+    class RequestValues(val clientId: String) : UseCase.RequestValues
+    class ResponseValue(
+        val invoiceList: List<Invoice?>
+    ) : UseCase.ResponseValue
+
+    override fun executeUseCase(requestValues: RequestValues?) {
```
Comment (therajanmaurya) [inline core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/invoice/FetchInvoices.kt]: This should not be nullable, Please remove `?`

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/invoice/presenter/InvoicePresenter.kt [hunk 2]
```diff
@@ -28,7 +28,7 @@ class InvoicePresenter @Inject constructor(
     }
 
     override fun getInvoiceDetails(data: Uri?) {
-        mUseCaseHandler.execute(fetchInvoiceUseCase, FetchInvoice.RequestValues(data),
+        mUseCaseHandler.execute(fetchInvoiceUseCase, data?.let { FetchInvoice.RequestValues(it) },
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/invoice/presenter/InvoicePresenter.kt]: Make above changes and remove let and pass directly `FetchInvoice.RequestValues(it)`

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/invoice/FetchInvoices.kt (hunk 1 above)
Comment (PratyushSingh07) [reply core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/invoice/FetchInvoices.kt]: done

**Comment by PratyushSingh07** [comment]
pushed the requested changes @therajanmaurya

**Pull request merged**